Zotero/Scenes/Detail/Lookup/ViewModels/LookupActionHandler.swift

```diff
@@ -106,82 +110,22 @@ final class LookupActionHandler: ViewModelActionHandler, BackgroundDbProcessingA
-                    state.lookupState = .failed(LookupState.Error.noIdentifiersDetected)
-                }
-                return
-            }
-
-            var lookupData = identifiers.map({ LookupState.LookupData(identifier: self.identifier(from: $0), state: .enqueued) })
-
-            self.update(viewModel: viewModel) { state in
-                if !state.multiLookupEnabled {
-                    state.lookupState = .lookup(lookupData)
-                } else {
-                    switch state.lookupState {
-                    case .lookup(let data):
-                        lookupData.append(contentsOf: data)
-                    default: break
-                    }
-
-                    state.lookupState = .lookup(lookupData)
-                }
-            }
-
-        case .item(let data):
-            guard let lookupId = data["identifier"] as? [String: String] else { return }
-            let identifier = self.identifier(from: lookupId)
-
-            if data.keys.count == 1 {
-                self.update(lookupData: LookupState.LookupData(identifier: identifier, state: .inProgress), in: viewModel)
-                return
-            }
-
-            if let error = data["error"] {
-                DDLogError("LookupActionHandler: \(identifier) lookup failed - \(error)")
-                self.update(lookupData: LookupState.LookupData(identifier: identifier, state: .failed), in: viewModel)
-                return
-            }
-
-            guard let itemData = data["data"] as? [[String: Any]], let item = itemData.first, let parsedData = self.parse(item, viewModel: viewModel, schemaController: self.schemaController) else {
-                self.update(lookupData: LookupState.LookupData(identifier: identifier, state: .failed), in: viewModel)
-                return
-            }
-
-            self.backgroundQueue.async { [weak self, weak viewModel] in
-                guard let self = self else { return }
-
-                do {
-                    try self.storeDataAndDownloadAttachmentIfNecessary(parsedData)
-
-                    inMainThread { [weak self] in
-                        guard let self = self, let viewModel = viewModel else { return }
-                        let translatedData = LookupState.LookupData(identifier: identifier, state: .translated(parsedData))
-                        self.update(lookupData: translatedData, in: viewModel)
-                    }
-                } catch let error {
-                    DDLogError("LookupActionHandler: can't create item(s) - \(error)")
-
-                    inMainThread { [weak self] in
-                        guard let self = self, let viewModel = viewModel else { return }
-                        let failedData = LookupState.LookupData(identifier: identifier, state: .failed)
-                        self.update(lookupData: failedData, in: viewModel)
-                    }
+                    state.lookupState = .lookup(data)
```

Comment:
Check https://github.com/zotero/zotero-ios/pull/707/commits/a29a864c86ff0eabc79981d05550e39492fdba52

To make things hopefullly simpler:
* A separate `LookupState` for `waitingInput` was added, and `loadingIdentifiers` is used as before just for waiting until the first updates arrive. 
* Since `IdentifierLookupController` needs to hold the lookup state anyway, it passes it to every update it publishes, so `LookupActionHandler` just needs to copy it when needed.
* Also fixed some potential edge case bugs in IdentifierLookupController, by making the workflow for publishing updates more strict.